pkg/asset/ignition/bootstrap/common.go

```diff
@@ -264,9 +265,16 @@ func (a *Common) getTemplateData(dependencies asset.Parents, bootstrapInPlace bo
 		platformData.VSphere = vsphere.GetTemplateData(installConfig.Config.Platform.VSphere)
 	}
 
+	bootstrapNodeIP := os.Getenv("OPENSHIFT_INSTALL_BOOTSTRAP_NODE_IP")
+	if bootstrapNodeIP != "" && net.ParseIP(bootstrapNodeIP) == nil {
+		logrus.Warnf("OPENSHIFT_INSTALL_BOOTSTRAP_NODE_IP must have valid ip address, given %s. Skipping it", bootstrapNodeIP)
+		bootstrapNodeIP = ""
+	}
+
 	var APIIntVIPonIPv6 bool
 	platformAPIVIP := apiVIP(&installConfig.Config.Platform)
-	if platformAPIVIP == "" {
+	// if specific ip was set for bootstrap we should use it
+	if platformAPIVIP == "" || bootstrapNodeIP != "" {
```

Comment:
It would seem more straightforward to me to use if/else in the template:

```
{{- if .BootstrapNodeIP }}
    --node-ip="{{.BootstrapNodeIP}}" \
    --address="{{.BootstrapNodeIP}}" \
{{- else}}
{{- if .UseIPv6ForNodeIP }}
    --node-ip=:: \
{{- end}}
{{- end}}
```

rather than trying to inhibit them both being true in the logic here, when the reason for it is not locally obvious.